Project-Hotpot/Project-Hotpot/Algorithm/SongAlgorithm.swift

```diff
@@ -35,42 +34,23 @@ class SongAlgorithm{
     }
     
     func getRandomGenre() -> String? {
-        let genre = api_instance.genreSeedArray.randomElement()
+        //returns nil if the genreArray is empty
+        guard let genre = api_instance.genreSeedArray.randomElement() else {
+            return nil
```

Comment:
I'm not seeing a use of `.isEmpty` here, so I believe this effectively becomes "return nil if genre is nil". Were you receiving an error at all here...?